_includes/tailwind-config.html

```diff
@@ -3,69 +3,33 @@
         theme: {
             extend: {
                 colors: {
-                    'primary': {
-                        '50': '#eff2fe',
-                        '100': '#e3e6fc',
-                        '200': '#ccd1f9',
-                        '300': '#acb3f5',
-                        '400': '#8b8bee',
-                        '500': '#776fe5',
-                        '600': '#6753d8',
-                        '700': '#5844be',
-                        '800': '#473a99',
-                        '900': '#3d357a', /* main */
-                        '950': '#251f47',
+                    primary: {
+                        DEFAULT: 'hsla(var(--primary-h), var(--primary-s), var(--primary-l), 1)',
+                        light: 'hsla(var(--primary-h), calc(var(--primary-s) - 12%), calc(var(--primary-l) + 58%), 1)',
+                        hover: 'hsla(var(--primary-h), var(--primary-s), calc(var(--primary-l) + 12%), 1)',
+                        'light-hover': 'hsla(var(--primary-h), calc(var(--primary-s) - 12%), calc(var(--primary-l) + 52%), 1)',
+                        content: 'hsla(var(--primary-h), var(--primary-s), calc(var(--primary-l) + 70%), 1)',
+                        'light-content': 'hsla(var(--primary-h), var(--primary-s), var(--primary-l), 1)',
```

Comment:
There are a lot of "magic numbers" here to calculate light, hover, etc. and they are different for primary, secondary, etc. Will these same numbers work for other choices of primary, secondary, etc. colors?

Looks like relative colors are cleaner but do not have perfect coverage in browsers yet. I wonder can we use them since Chrome, Firefox, Edge, Safari all support it? https://caniuse.com/css-relative-colors That would allow this type of thing (from stackoverflow first answer here https://stackoverflow.com/questions/55329996/how-to-create-color-shades-using-css-variables-similar-to-darken-of-sass).

```
  --color-primary: #f00; /* any format you want here */
  --color-primary-darker: hsl(from var(--color-primary) h s calc(l - 5));
  --color-primary-darkest: hsl(from var(--color-primary) h s calc(l - 10));
```

Reply:
> There are a lot of "magic numbers" here to calculate light, hover, etc. and they are different for primary, secondary, etc. Will these same numbers work for other choices of primary, secondary, etc. colors?
> 
> Looks like relative colors are cleaner but do not have perfect coverage in browsers yet. I wonder can we use them since Chrome, Firefox, Edge, Safari all support it? https://caniuse.com/css-relative-colors That would allow this type of thing (from stackoverflow first answer here https://stackoverflow.com/questions/55329996/how-to-create-color-shades-using-css-variables-similar-to-darken-of-sass).
> 
> ```
>   --color-primary: #f00; /* any format you want here */
>   --color-primary-darker: hsl(from var(--color-primary) h s calc(l - 5));
>   --color-primary-darkest: hsl(from var(--color-primary) h s calc(l - 10));
> ```

I was unable to achieve something I was happy with using relative color. I agree that there were too many magic numbers in there, especially with them needing to be different based on the color. The goal is for the user to just plug in their color and have the other shades and states handled for them. So I opted for a js approach to calculate colors. For simplicity, I nixed the "light" versions of the colors since we already have ghost buttons for them (we don't need so many variations).

Now the user can just put their hex values in the CSS variables and be done with it.